sklearn_numba_dpex/testing/config.py

```diff
@@ -0,0 +1,14 @@
+from contextlib import contextmanager
+
+
+@contextmanager
+def override_attr_context(obj, **attrs):
+    attrs_before = dict()
+    for attr_name, attr_value in attrs.items():
+        attrs_before[attr_name] = getattr(obj, attr_name)
+        setattr(obj, attr_name, attr_value)
```

Comment:
```suggestion
        attrs_before[attr_name] = getattr(obj, attr_name, default=None)
        setattr(obj, attr_name, attr_value)
```

Reply:
In fact not inputing a default is in purpose here, the context manager is called `override` because I'd like it to be used only to override existing attributes, and raising `AttributeError` if trying to override an attribute that does not exist.

(also, it's simpler this way, else to clean properly in the exit we should track attributes that exist and those who don't...)

I'll add a docstring to make this more understandable.